lib/recurly/transaction.rb

```diff
@@ -125,5 +125,36 @@ def inspect
       end
       super attributes
     end
+
+    # https://github.com/recurly/recurly-client-ruby/pull/436
+    def marshal_dump
+      [
+        @attributes.reject { |k, v| v.is_a?(Proc) },
```

Comment:
Could we change this definition to use the code from the resource class if possible?

```ruby
def marshal_dump
  super << @type
end
```

Reply:
@aaron-suarez do we need this implementation at all?

Reply:
I thought that too but I felt it is clearer to have those methods displayed together on the class.

Reply:
My assumption was that we don't need to override these methods at all. We should be able to add `@type` to both marshal_dump and marshal_load. That would just mean that every `Resource` would have a `@type` instance variable but only the ones that use that xml attribute will have the `attr_reader :type` so it won't be exposed in resources where it's always nil (and it would be nil since extra variables in a destructuring statement are set to nil). We'd probably want to change places that look for the @type to be defined to instead ask if the object `respond_to?(:type)`. Example:
https://github.com/recurly/recurly-client-ruby/blob/d5d5c7bac5ee5308c3b2f38aa1d758ca9bf30acb/lib/recurly/resource.rb#L1027

I'm okay with the current path the code is taking too. I haven't tested this and there might be things I'm missing. I'm just providing some feedback but the final decision is for @aaron-suarez . Let's do whatever works 👍